vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -0,0 +1,140 @@
+      new_result
+    end
+
+    def parse_node(node)
+      new_result = parse_base_item(node)
+      # TODO, add more properties such as resources
+      new_result
+    end
+
+    def parse_service(service)
+      new_result = parse_base_item(service)
+      new_result[:port] = service.spec.port
+      new_result[:protocol] = service.spec.protocol
+      new_result[:portal_ip] = service.spec.portalIP
+      new_result[:container_port] = service.spec.containerPort
+      new_result[:namespace] = service.metadata.instance_values["table"][:namespace]
+      new_result[:session_affinity] = service.spec.sessionAffinity
+      new_result
+    end
+
+    def parse_pod(pod)
+      # pod in kubernetes is container group in manageiq
+      new_result = parse_base_item(pod)
+      # namespace is overriden in more_core_extensions and hence needs a non method access
+      new_result[:namespace] = pod.metadata.instance_values["table"][:namespace]
+      # workaround due to https://github.com/GoogleCloudPlatform/kubernetes/issues/3607
+      new_result[:restart_policy] = pod.spec.restartPolicy.to_h.keys.first.to_s
+      new_result[:dns_policy] = pod.spec.dnsPolicy
+      new_result[:container_definitions] = []
+      # TODO, map volumes
+      # TODO, podIP
+      containers = pod.spec.containers
+      containers.each do |container_def|
+        container_def = parse_container_definition(container_def, pod.metadata.uid)
+        new_result[:container_definitions] << container_def
+      end
+
+      # container instances
+
+      unless pod.status.info.nil?
+        new_result[:containers] = []
+        pod.status.info.to_h.each do |container_name, container|
+          container = parse_container(container, container_name, pod.metadata.uid)
+          new_result[:containers] << container
+        end
+      end
+      new_result
+    end
+
+    def parse_container_definition(container_def, pod_id)
+      new_result = {}
+      new_result[:ems_ref] = "#{pod_id}_#{container_def["name"]}_#{container_def["image"]}"
+      new_result[:name] = container_def["name"]
+      new_result[:image] = container_def["image"]
+      new_result[:image_pull_policy] = container_def["imagePullPolicy"]
+      new_result[:memory] = container_def["memory"]
+      # to_i already removes the 'm'
+      # https://github.com/GoogleCloudPlatform/kubernetes/blob/0b801a91b15591e2e6e156cf714bfb866807bf30/pkg/api/v1beta3/types.go#L815
+      new_result[:cpu_cores] = container_def["cpu"].to_f / 1000
+      new_result[:container_port_configs] = []
+      ports = container_def["ports"]
+      ports.each do |port_entry|
+        port_config = parse_container_port_config(port_entry, pod_id, container_def["name"])
+        new_result[:container_port_configs] << port_config
+      end
+      new_result
+    end
+
+    def parse_container(container, container_name, pod_id)
+      new_result = {}
+      new_result[:ems_ref] = "#{pod_id}_#{container_name}_#{container["image"]}"
```

Comment:
I think this is good, but I'm wondering if a multi-key might be better in the save_inventory.